src/brain-map.ts

```diff
@@ -0,0 +1,30 @@
+import { GameState } from './game-state';
+
+export class BrainMap {
+  state: GameState[] = new Array(20000);
+
+  constructor(initState: GameState[] = []) {
+    initState.forEach((move, i) => {
+      this.state[i] = move;
+    });
+  }
+
+  set(key: string, state: GameState) {
+    this.state[this.createHash(key)] = state;
+  }
+
+  get(key: string) {
+    return this.state[this.createHash(key)];
+  }
+
+  createHash(key: string) {
+    let hash = 5381;
+    let i = key.length;
+
+    while (i) {
+      hash = (hash * 33) ^ key.charCodeAt(--i);
+    }
+
+    return (hash >>> 0) % 20000;
```

Comment:
So I'm not familiar with the `>>>` operator, what does that do?

Reply:
I honestly don't know. I took it from the another hash function implementation

Reply:
Haha, okay, I know that `>>` is a bit shift. So it looks like `>>>` is a `0` fill bit shift to the right. So you are shifting the binary representation of hash to the right 0 bits. So nothing.

Example: 5 >>> 2 would be:
5: `00000101` shifting to the right two `00000000`

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators